libraries/c_sdk/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -1149,7 +1149,7 @@ IotMqttError_t IotMqtt_Connect( const IotMqttNetworkInfo_t * pNetworkInfo,
         transport.recv = transportRecv;
 
         /* Fill the values for network buffer. */
-        networkBuffer.pBuffer = &( connToContext[ contextIndex ].buffer );
+        networkBuffer.pBuffer = &( connToContext[ contextIndex ].buffer[0] );
```

Comment:
Was this incorrect way of getting the starting address not causing issues in testing before?

Reply:
This was not extensively tested, However this didn't cause any issue in tests on WinSim. It is good to catch the fix through a build warning reported as error.